Sources/Classes/RSEventRepository.m

```diff
@@ -106,39 +107,41 @@ - (void) setAnonymousIdToken {
 }
 
 - (void) __initiateSDK {
+    __weak RSEventRepository *weakSelf = self;
     dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+        RSEventRepository *strongSelf = weakSelf;
         int retryCount = 0;
-        while (self->isSDKInitialized == NO && retryCount <= 5) {
-            RSServerConfigSource *serverConfig = [self->configManager getConfig];
-            int receivedError =[self->configManager getError];
+        while (strongSelf->isSDKInitialized == NO && retryCount <= 5) {
```

Comment:
@desusai7 Quick check as I might have forgotten.. if strongSelf is null this line won't crash. If yes, let's add a null check

Reply:
@anujkaura No need as if it is null, it will not execute.